addons/event_crm/models/event_lead_generate_request.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class EventLeadGenerateRequest(models.Model):
+    """ Technical model created when a user requests 'leads generation' on an event based on all
+    existing event.lead.rules (see event#action_generate_leads).
+
+    As an event can hold a lot of registrations, we use a batch approach with a separate model that
+    contains the batching logic method and the field to retain progress.
+
+    To benefit from a background processing, we use a CRON that calls itself with a CRON trigger
+    until the batch is completed, which unlinks this technical generation record. """
+
+    _name = "event.lead.generate.request"
+    _description = "Event Lead Generate Request"
+    _log_access = False
+
+    _REGISTRATIONS_BATCH_SIZE = 200
+
+    event_id = fields.Many2one('event.event', required=True, string="Event")
+    processed_registration_id = fields.Integer("Processed Registration",
+        help="The ID of the last processed event.registration, used to know where to resume.")
+
+    @api.model
+    def _cron_generate_leads(self):
+        """ See class docstring for details. """
+        generate_requests = self.env['event.lead.generate.request'].search([])
```

Comment:
Would probably give a limit to the number of requests we want to run, in case it was spammed somehow
```suggestion
        generate_requests = self.env['event.lead.generate.request'].search([], limit=100)
```